protocol.md

```diff
@@ -837,6 +837,71 @@ interface TextDocumentContentChangeEvent {
 }
 ```
 
+#### Content Request
+
+The content request is sent from the server to the client to request the current content of any text document.
+This allows language servers to operate without using the file system directly.
+
+_Request_:
+* method: 'textDocument/content'
+* params: `ContentRequestParams` defined as follows:
+
+```typescript
+interface ContentRequestParams {
+	/**
+	 * The text document to receive the content for
+	 */
+	textDocument: TextDocumentIdentifier;	
+}
+```
+
+_Response_:
+* result: `string`
+* error: code and message set in case an exception happens during showing a message.
+
+#### Glob Request
+
+The glob request is sent from the server to the client to request a list of files in the workspace that match a glob pattern.
+The glob pattern must be matched only against the path of the files in the workspace and relative to the `rootPath`.
```

Comment:
Lots of LS need to fetch files outside the `rootPath`. For example, the rootPath might be file:///gopath/src/github.com/foo/bar but Go needs to fetch from the GOROOT (/usr/local/go/src), Python might need to fetch from /usr/lib/python/whatever, etc. You could specify a pattern of `../../../**/*.php` but it's also not clear if those files are "in the workspace."